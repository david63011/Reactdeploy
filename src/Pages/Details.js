import { useDispatch, useSelector } from 'react-redux';
import { fetchDetail } from '../Redux/DetailSlice';
import { useEffect } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import back2 from '../images/back2.png';
const Details = () => {
  const navigate = useNavigate();
  const { id } = useParams();
  const dispatch = useDispatch();
  const crcdetail = useSelector((state) => state.detail);
  useEffect(() => {
    dispatch(fetchDetail(id));
  });
  return (
    <div className="big">
      <h1 className="dheader">The Crypto Market</h1>
      <div className="detailcontainer">
        <img
          onClick={() => navigate(-1)}
          className="img-money"
          src={back2}
          alt="/"
        ></img>
        <div className="imageheader">
          <h1 className="detailname">{crcdetail.name}</h1>
          <img className="detailimg" src={crcdetail.icon} alt="crypto" />
        </div>
        <div className="detailList">
          <ul>
            <li>Ranking: {crcdetail.rank}</li>
            <li>The Current Price $: {crcdetail.price}</li>
            <li>Market Cap: {crcdetail.marketCap}</li>
            <li>Volume: {crcdetail.volume}</li>
            <li>Total Supply: {crcdetail.totalSupply}</li>
            <li>Available Supply: {crcdetail.availableSupply}</li>
            <li>1 day price change: {crcdetail.priceChange1d}</li>
            <li>1 week price change: {crcdetail.priceChange1w}</li>
          </ul>
        </div>
      </div>
      ;
    </div>
  );
};

export default Details;
