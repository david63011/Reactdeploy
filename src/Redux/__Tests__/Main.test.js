import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Details from '../../Pages/Details';
import configureStore from '../../Redux/ConfigureStore';

test('Main renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <Details />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
