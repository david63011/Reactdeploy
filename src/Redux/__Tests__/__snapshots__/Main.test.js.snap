// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div>
  <div
    className="NavBar"
  >
    <h1
      className="logoname"
    >
      Crypto Market
    </h1>
    <div
      className="icons"
    >
      <img
        alt="/"
        className="img-money"
        src="bull.png"
      />
      <img
        alt="/"
        className="img-money"
        src="dollars.png"
      />
      <img
        alt="/"
        className="img-money"
        src="oso.png"
      />
    </div>
  </div>
  <div
    className="coins"
  >
    <div
      className="coins-container"
    >
      <div
        className="inputContainer"
      >
        <img
          alt="/"
          className="img-search"
          src="search.png"
        />
        <input
          className="textInput"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
